python/cudf/cudf/core/buffer.py

```diff
@@ -1,218 +1,83 @@
-import pickle
-import types
+import functools
+import operator
 
 import numpy as np
 
 import rmm
+from rmm import DeviceBuffer
 
-from cudf.utils import cudautils, utils
 
-
-class Buffer(object):
-    """A 1D gpu buffer.
-    """
-
-    _cached_ipch = None
-
-    @classmethod
-    def from_empty(cls, mem, size=0):
-        """From empty device array
-        """
-        return cls(mem, size=size, capacity=mem.size)
-
-    @classmethod
-    def null(cls, dtype):
-        """Create a "null" buffer with a zero-sized device array.
+class Buffer:
```

Comment:
Should consider moving this whole class to RMM, but we can do that later.

Reply:
Being discussed in issue ( https://github.com/rapidsai/rmm/issues/227 ).